src/main/java/seedu/address/model/contact/TsContainsKeywordsPredicate.java

```diff
@@ -18,11 +18,15 @@ public TsContainsKeywordsPredicate(List<String> techKeywords) {
 
     @Override
     public boolean test(Contact contact) {
-        return techKeywords.stream().allMatch(keyword ->
-            contact.getTechStack().stream().anyMatch(ts ->
-                StringUtil.containsWordIgnoreCase(ts.techStackName, keyword)
-            )
-        );
+        if (techKeywords.isEmpty()) {
+            return false;
+        } else {
+            return techKeywords.stream().allMatch(keyword ->
+                    contact.getTechStack().stream().anyMatch(ts ->
+                            StringUtil.containsWordIgnoreCase(ts.techStackName, keyword)
+                    )
+            );
+        }
```

Comment:
Else block is technically not needed
```suggestion
        if (techKeywords.isEmpty()) {
            return false;
        }
        return techKeywords.stream().allMatch(keyword ->
                contact.getTechStack().stream().anyMatch(ts ->
                        StringUtil.containsWordIgnoreCase(ts.techStackName, keyword)
                )
        );
```